federation-js/src/composition/compose.ts

```diff
@@ -514,6 +523,7 @@ export function addFederationMetadataToSchemaNodes({
       // TODO: Why don't we need to check for non-object types here
       if (isObjectType(namedType)) {
         const field = namedType.getFields()[fieldName];
+        if (!field) break;
```

Comment:
As seen [here](https://github.com/apollographql/federation/pull/478/files#diff-9ae375eec1a337fb4d06139755b68ec0d4fe9d6168b0f207c29e6f2a31882daeR193) in the new test case, when a type goes unrecognized, fields which return that type are abandoned. This handles the case where a field which is expected to exist cannot actually be found in the schema.